### ISSUE TITLE
Rework the Hadoop windows command

### ISSUE THREAD

**Issue by ilija-lazoroski** [guardicore/monkey#1781]
## Describe the bug
Hadoop in its `nature` is repulling the monkey binaries which causes issues when trying to run the exploiter.
In order for the windows machine to be exploited we need to remove the binary and rerun it again. 

## To Reproduce
1. Configure the Monkey to exploit windows Hadoop
2. Run the monkey on the hadoop windows machine
3. See that is shows exploited but no monkey runs

## Expected behavior
The hadoop windows command should check if there is a binary and that is not empty,
remove it and download the new one. See this [commit](https://github.com/guardicore/monkey/pull/1765/commits/c802f217565f1ad29b750174099974fa3d2cec8a)

## Machine version (please complete the following information):
 - OS: Windows

## Tasks
- [ ] Run a test on Windows/Hadoop. Verify that all agent binaries are removed after running. If not, fix it. (0.25d) @ilija-lazoroski 
- [x] Run a test on Linux/Hadoop. Verify that all agent binaries are removed after running. If not, fix it. (0d) @ilija-lazoroski

**Comment by mssalvatore** [comment]
This should be mostly resolved by https://github.com/guardicore/monkey/issues/1782

**Comment by ilija-lazoroski** [comment]
Hadoop uses [speculative exection](https://data-flair.training/blogs/speculative-execution-in-hadoop-mapreduce/) which separate a job into two tasks which runs in parallel which means that our exploit command which downloads and executes the monkey will be run twice (not always the case), and may leave the executable on the machine. In linux that was fixed in the hadoop exploit command, as when we requested another file we were left with zero sized one. In windows there is not similar mechanism to the linux command. The solution to this issue was to mark the agent file for deletion on boot and try to delete it. This is going to fix any issue that we may have with future exploits.
